src/parallel_simulation.rs

```diff
@@ -110,12 +130,46 @@ impl Simulation {
         info!("Partition #{} finished.", self.scenario.msg_broker.id,);
     }
 
-    fn wakeup(&mut self, now: u32) {
+    fn wakeup(&mut self, now: u32, router: &mut Option<Router>) {
         let agents_2_link = self.activity_q.wakeup(now);
 
         for id in agents_2_link {
             let agent = self.scenario.population.agents.get_mut(&id).unwrap();
             self.events.handle_act_end(now, &agent);
+
+            if router.is_some() {
```

Comment:
I think `if let Some(mut router) = router` would be more idiomatic. You get rid of the `router.as_mut().unwrap()` on line 152.